enocean/__init__.py

```diff
@@ -628,49 +711,118 @@ def send_rgbw_dim(self,id_offset=0, color='red', dim=0, dimspeed=0):
+            self.logger.error("enocean: send_switch(): error")
+            return None
+        # Send command
+        #self.logger.info('enocean: sending switch command A5_38_08_01')
+        self.logger.info('enocean: send_switch(): command with [ID-Offset], [RORG], [payload] / [{:#04x}], [{:#04x}], [{}]'.format(id_offset, rorg, ', '.join('{:#04x}'.format(x) for x in payload)))
+        self._send_radio_packet(id_offset, rorg, payload)
+
 
+#################################################
+### --- END - Definitions of Send Methods --- ###
+#################################################
+
+
+####################################################
+### --- START - Definitions of Learn Methods --- ###
+####################################################
     def send_learn_dim(self, id_offset=0):
+        self.logger.debug('enocean: send_learn_dim()')
+        # Prepare Data
+        rorg = 0xA5
+        # Only for Eltako FSUD-230V
+        payload = [0x02, 0x00, 0x00, 0x00]
+        # For other Eltako Dim Devices use this payload
+        #payload = [0xE0, 0x40, 0x0D, 0x80]
+
         if (id_offset < 0) or (id_offset > 127):
-            self.logger.error("enocean: ID offset out of range (0-127). Aborting.")
-            return
-        self.logger.info("enocean: sending learn telegram for dim command")
-        self._send_radio_packet(id_offset, 0xA5, [0x02, 0x00, 0x00, 0x00])
+            self.logger.error('enocean: ID offset with value = {} out of range (0-127). Aborting.'.format(id_offset))
+            return None
+        self.logger.info('enocean: sending learn telegram for dim command with [ID-Offset], [RORG], [payload] / [{:#04x}], [{:#04x}], [{}]'.format(id_offset, rorg, ', '.join('{:#04x}'.format(x) for x in payload)))
+        self._send_radio_packet(id_offset, rorg, payload)
+
 
     def send_learn_rgbw_dim(self, id_offset=0):
+        self.logger.debug('enocean: send_learn_rgbw_dim()')
+        # Prepare Data
+        rorg = 0xA5
+        payload = [0xFF, 0xF8, 0x0D, 0x87]
+
         if (id_offset < 0) or (id_offset > 127):
-            self.logger.error("enocean: ID offset out of range (0-127). Aborting.")
-            return
-        self.logger.info("enocean: sending learn telegram for rgbw dim command")
-        self._send_radio_packet(id_offset, 0x07, [0xFF, 0xF8, 0x0D, 0x87])
+            self.logger.error('enocean: ID offset with value = {} out of range (0-127). Aborting.'.format(id_offset))
+            return None
+        self.logger.info('enocean: sending learn telegram for rgbw dim command with [ID-Offset], [RORG], [payload] / [{:#04x}], [{:#04x}], [{}]'.format(id_offset, rorg, ', '.join('{:#04x}'.format(x) for x in payload)))
+        self._send_radio_packet(id_offset, rorg, payload)
+
 
     def send_learn_switch(self, id_offset=0):
+        self.logger.debug('enocean: send_learn_switch()')
+        # Prepare Data
+        rorg = 0xA5
+        payload = [0xE0, 0x40, 0x0D, 0x80]
```

Comment:
Looks right. Source: https://www.eltako.com/fileadmin/downloads/de/Gesamtkatalog/Eltako_Gesamtkatalog_KapT_low_res.pdf. Approved, closing this one.